src/request-pipeline/index.js

```diff
@@ -90,6 +90,11 @@ var stages = {
 function createReqOpts (ctx) {
     var bodyWithUploads = injectUpload(ctx.req.headers['content-type'], ctx.reqBody);
 
+    if (bodyWithUploads)
```

Comment:
We need comment here about length recalculation. Because, initially it was done this way, but then removed, because it's not clear from code why we can't do it inline